src/main.rs

```diff
@@ -87,7 +97,13 @@ async fn new(cookies: &CookieJar<'_>, request: Json<NewRequest>) -> Result<Strin
 		)
 		.map_err(|_| Status::InternalServerError)?;
 		// 3. add to storage
-		store::write(mmr_store.clone());
+		if USE_DB {
+			let final_object_id = store::write_to_db(DB_NAME, COLLECTION_NAME, mmr_store.clone(), db, None).await;
+			cookies.add(Cookie::new("object_id", final_object_id.clone()));
+		} else {
+			store::write_to_file(mmr_store.clone());
+		}
+		
```

Comment:
Let's remove the `write_to_file` fn, it was supposed to be a mock only, as it wouldn't work with concurrent wallets.
The endpoint shouldn't know anything about the store, whether it is a file, a db or whatever should be abstracted away. Let's move `DB_NAME` and `COLLECTION_NAME` to `store.rs`, also let's remove the `None` parameter.
Finally, the parameter `db` shouldn't be here either and should be encapsulated in `store.rs`

Reply:
I do think that abstracting the data access layer is a good thing to do, however I believe that we should make another story for this. The way that I have set up the database connection is via the docs here: https://rocket.rs/guide/v0.5/state/ and here: https://api.rocket.rs/v0.5/rocket_db_pools/ where Rocket is managing the DB pool for us. Note that I have to attach the DB pool during rocket initialization hence why the db variable is in main.rs.

Reply:
Code examples that you can find out there, even official ones, are just that, code examples that show you how one can use their software in a simple way.
The fact that something needs to be attached at initialization doesn't mean that it has to be a MongoDB, it can be the abstracted data layer itself, that's what I propose to do here https://github.com/ideal-lab5/murmur-api/blob/9413a8750f731d03b5c692bcefaf90c360d886c1/src/main.rs#L162-L165

Reply:
Yes I know they're just examples. I was looking at the API docs about implementing custom code and it looked like a lot of work. I have taken your changes and implemented them.